mutable_batch/src/payload/partition.rs

```diff
@@ -846,6 +1082,28 @@ mod tests {
         }
     }
 
+    prop_compose! {
+        /// Yield a Vec containing an identical timestamp run of random length,
+        /// up to `max_run_len`,
+        fn arbitrary_timestamp_run(max_run_len: usize)(v in 0_i64..i64::MAX, run_len in 1..max_run_len) -> Vec<i64> {
+            let mut x = Vec::with_capacity(run_len);
+            x.resize(max_run_len, v);
```

Comment:
WAIT this line is always resizing the vec to have 6 elements, so that's why there's that lower bound! I think this should be:



```suggestion
            x.resize(run_len, v);
```